lib/chore/queues/sqs/publisher.rb

```diff
@@ -3,51 +3,64 @@
 module Chore
   module Queues
     module SQS
-      
       # SQS Publisher, for writing messages to SQS from Chore
       class Publisher < Chore::Publisher
         @@reset_next = true
 
+        # @param [Hash] opts Publisher options
         def initialize(opts={})
           super
           @sqs_queues = {}
           @sqs_queue_urls = {}
         end
 
-        # Takes a given Chore::Job instance +job+, and publishes it by looking up the +queue_name+.
+        # Publishes a message to an SQS queue
+        #
+        # @param [String] queue_name Name of the SQS queue
+        # @param [Hash] queue_name Job instance definition, will be encoded to JSON
+        #
+        # @return [struct Aws::SQS::Types::SendMessageResult]
         def publish(queue_name,job)
-          queue = self.queue(queue_name)
-          queue.send_message(encode_job(job))
+          queue = queue(queue_name)
+          queue.send_message(message_body: encode_job(job))
         end
 
         # Sets a flag that instructs the publisher to reset the connection the next time it's used
+        # TODO: We reset on every publish. Is this desirable? Seems like a performance problem.
```

Comment:
Calling this out. I don't think we should fix it in this PR, but I'd like to know if we should leave the TODO in.

Reply:
Can you point to where this is reset on every *publish* (just so we're using the same vocab, "publish" meaning writing a message to the sqs or filesystem queue).  The only place I see us resetting the connection is after forking, which is required.

Reply:
We are setting `@@reset_next = true` every time `reset_connection!` is called

https://github.com/Tapjoy/chore/blob/04cd7a8745de51403f20ca9fa8b6f209b3bdce7b/lib/chore/queues/sqs/publisher.rb#L32-L35

Which means we will reset the connection every time `queue` is called

https://github.com/Tapjoy/chore/blob/04cd7a8745de51403f20ca9fa8b6f209b3bdce7b/lib/chore/queues/sqs/publisher.rb#L55-L60

And we call `queue` on every publish

https://github.com/Tapjoy/chore/blob/04cd7a8745de51403f20ca9fa8b6f209b3bdce7b/lib/chore/queues/sqs/publisher.rb#L23-L26

Reply:
Aaaand now I see the `@@reset_next = false` ... so nevermind! My brain looked right past that.